tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedEventMatcher.cs

```diff
@@ -0,0 +1,168 @@
+                        {
+                            switch (element.Name)
+                            {
+                                case "command":
+                                    _valueMatcher.AssertValuesMatch(commandStartedEvent.Command, element.Value);
+                                    break;
+                                case "commandName":
+                                    commandStartedEvent.CommandName.Should().Be(element.Value.AsString);
+                                    break;
+                                case "databaseName":
+                                    commandStartedEvent.DatabaseNamespace.DatabaseName.Should().Be(element.Value.AsString);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unexpected commandStartedEvent field: '{element.Name}'.");
+                            }
+                        }
+                        break;
+                    case "commandSucceededEvent":
+                        var commandSucceededEvent = actualEvent.Should().BeOfType<CommandSucceededEvent>().Subject;
+                        foreach (var element in expectedEventValue)
+                        {
+                            switch (element.Name)
+                            {
+                                case "reply":
+                                    _valueMatcher.AssertValuesMatch(commandSucceededEvent.Reply, element.Value);
+                                    break;
+                                case "commandName":
+                                    commandSucceededEvent.CommandName.Should().Be(element.Value.AsString);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unexpected commandStartedEvent field: '{element.Name}'.");
+                            }
+                        }
+                        break;
+                    case "commandFailedEvent":
+                        var commandFailedEvent = actualEvent.Should().BeOfType<CommandFailedEvent>().Subject;
+                        foreach (var element in expectedEventValue)
+                        {
+                            switch (element.Name)
+                            {
+                                case "commandName":
+                                    commandFailedEvent.CommandName.Should().Be(element.Value.AsString);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unexpected commandStartedEvent field: '{element.Name}'.");
+                            }
+                        }
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized event type: '{expectedEventType}'");
```

Comment:
This exception was missed when adding periods to the end of exception messages:

    throw new FormatException($"Unrecognized event type: '{expectedEventType}'.");

Reply:
My bad,
Done